src/components/ToggleElementDropdown/ToggleElementDropdown.js

```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import core from 'core';
+
+import selectors from 'selectors';
+import actions from 'actions';
+
+import { zoomTo } from 'helpers/zoom';
+
+import ToggleElementButton from '../ToggleElementButton';
+
+import './ToggleElementDropdown.scss';
+
+class ToggleElementDropdown extends React.PureComponent {
+
+  constructor(props){
+    super(props);
+    this.state = { zoom: Math.ceil(core.getZoom() * 100), value: 100 };
+    this.textarea = React.createRef();
+  }
+  componentDidMount(){
+    core.addEventListener('zoomUpdated', this.onZoomUpdated);
+  }
+
+  onZoomUpdated = () => {
+    this.setState({ zoom: Math.ceil(core.getZoom() * 100)},
+    ()=>{this.setState({ value: this.state.zoom })});
+  }
```

Comment:
I think instead of updating `this.state.value` after `this.state.value` is updated, we can do something like:
```javascript
// get zoom value somehow
this.setState({ value:  Math.ceil(zoom * 100) });
```

Reply:
Since zoom needs to be fetched from the core, should I modify the code to `this.setState({ value: Math.ceil(core.getZoom() * 100) });`?